apps/chat/src/components/Chat/ChatHeader.tsx

```diff
@@ -202,7 +210,10 @@ export const ChatHeader = ({
                         'cursor-not-allowed',
                     )}
                     disabled={isIsolatedView || isMessageStreaming}
-                    onClick={() => onModelClick(conversation.id)}
+                    onClick={() =>
+                      isDisallowChangeAgentEnabled &&
+                      onModelClick(conversation.id)
+                    }
```

Comment:
see comment above, it can be reverted because `isChangeAgentDisallowed` should disabled button